server/sonar-ce/src/main/java/org/sonar/ce/httpd/CeHttpServer.java

```diff
@@ -84,21 +96,49 @@ String getUrl() {
     return "http://" + nanoHttpd.getHostname() + ":" + nanoHttpd.getListeningPort();
   }
 
-  private class SystemInfoNanoHttpd extends NanoHTTPD {
+  private class CeNanoHttpd extends NanoHTTPD {
+    private final NanoHTTPD.Response response404 = newFixedLengthResponse(NOT_FOUND, MIME_PLAINTEXT, "Error 404, not found.");
+
+    private final ActionRegistryImpl actionRegistry;
 
-    SystemInfoNanoHttpd(String hostname, int port) {
+    CeNanoHttpd(String hostname, int port, ActionRegistryImpl actionRegistry) {
       super(hostname, port);
+      this.actionRegistry = actionRegistry;
     }
 
     @Override
     public Response serve(IHTTPSession session) {
-      ProtobufSystemInfo.SystemInfo.Builder infoBuilder = ProtobufSystemInfo.SystemInfo.newBuilder();
-      for (SystemInfoSection sectionProvider : sectionProviders) {
-        ProtobufSystemInfo.Section section = sectionProvider.toProtobuf();
-        infoBuilder.addSections(section);
+      return actionRegistry.getAction(session)
+        .map(action -> serveFromAction(session, action))
+        .orElse(response404);
+    }
+
+    private Response serveFromAction(IHTTPSession session, HttpAction action) {
+      try {
+        return action.serve(session);
+      } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)